release-calendar/src/client/api-service.ts

```diff
@@ -14,29 +14,35 @@
  * limitations under the License.
  */
 
-import {
-  ApiServiceInterface,
-  Promotion as PromotionEntity,
-  Release as ReleaseEntity,
-} from '../types';
-import {CurrentReleases, Release} from './models/view-models';
+import {ApiServiceInterface, Promotion} from '../types';
+import {CurrentReleases, EventInput} from './models/view-models';
 import fetch from 'node-fetch';
 const SERVER_URL = `http://localhost:3000`;
 
 export class ApiService implements ApiServiceInterface {
-  private getReleaseRequest(url: string): Promise<ReleaseEntity[]> {
+  private getPromotionRequest(url: string): Promise<Promotion[]> {
     return fetch(url).then((result) => result.json());
   }
-
-  private getPromotionRequest(url: string): Promise<PromotionEntity[]> {
+  private getPromotionRequests(url: string): Promise<Promotion[][]> {
     return fetch(url).then((result) => result.json());
   }
 
-  async getReleases(): Promise<Release[]> {
-    const releases = await this.getReleaseRequest(SERVER_URL);
-    return releases.map((release: ReleaseEntity) => {
-      return new Release(release);
+  async getReleases(): Promise<EventInput[]> {
+    const promotions = await this.getPromotionRequests(SERVER_URL);
+    const collect: EventInput[] = [];
+    promotions.forEach((promotionsInChannel: Promotion[]) => {
+      const endDate: Date = new Date();
+      collect.push(new EventInput(promotionsInChannel[0], endDate));
+      for (let i = 1; i < promotionsInChannel.length; i++) {
```

Comment:
I'm confused about this... it looks like you're iterating `promotionsInChannel` inside `promotionsInChannel`, so you'll have a lof of duplicates, wouldn't you?

Reply:
Hm, `promotions` is originally a 2d array which I'm making a 1d array out of. So `promotionsInChannel` is a row of promotions. Once I transform a single promotion that's from whatever row into a EventInput, I push it onto `collect`, the 1d array i'm making.

Reply:
Ah! Yes, the new variable names definitely help!